### PR TITLE
Super Rage Gives Ravager Bloodborne/Life Leech Slashing

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

If Rage is used while at less than 0 HP, the Ravager will regain HP whenever it slashes a living target equal to 50% to 100% of the damage dealt depending on Rage Power (how low the Ravager's HP was when Rage was used).

## Why It's Good For The Game

Makes Raging and fighting while at critical HP actually worth the risk outside of strict 1 v 1 scenarios because you now might actually not get instantly deleted via life leech.

## Changelog
:cl:
balance: If Rage is used while at less than 0 HP, the Ravager will regain HP whenever it slashes a living target equal to 50% to 100% of the damage dealt depending on Rage Power (how low the Ravager's HP was when Rage was used).
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
make it so rav can't passively heal when in endure, then

**Comment by Surrealaser** [comment]
> make it so rav can't passively heal when in endure with rage on, then. As Ravager with this would both passively heal and have leech.

Passive healing is basically irrelevant outside of resting

**Comment by KrabSpider** [comment]
> > make it so rav can't passively heal when in endure with rage on, then. As Ravager with this would both passively heal and have leech.
> 
> Passive healing is basically irrelevant outside of resting

What about healing infusion?

**Comment by Surrealaser** [comment]
> > > make it so rav can't passively heal when in endure with rage on, then. As Ravager with this would both passively heal and have leech.
> > 
> > 
> > Passive healing is basically irrelevant outside of resting
> 
> As for healing infusion?

Only applies during defensive fighting (i.e. you need to be on weeds), and is disabled by being on fire. Also the HP regained per tick isn't huge.

**Comment by Ragecrabbo** [comment]
Fun flavourful addition to rage, dont think it will have too much of an impact as you basically need to be on the brink of death to use it and the health you get from a slash will be equal to one or maybe two marine shots on you.

**Comment by SplinterGP** [comment]
i don't see why this is much needed or how its balanced, rage already does alot of stuff like sunder and something, what are the needs for it ?

**Comment by Surrealaser** [comment]
> i don't see why this is much needed or how its balanced, rage already does alot of stuff like sunder and something, what are the needs for it ?

I think I was abundantly clear.

At the moment it is basically suicide to try to benefit from Rage at sub-zero health; this makes the risk/reward a little more viable.

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 1]
```diff
@@ -352,6 +353,27 @@
 	to_chat(owner,"<span class='highdanger'>Our rage begins to subside... [ability_name] will only last for only [(RAVAGER_RAGE_DURATION + bonus_duration) * (1-RAVAGER_RAGE_WARNING) * 0.1] more seconds!</span>")
 	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
 
+///Warns the user when his rage is about to end.
+/datum/action/xeno_action/rage/proc/drain_slash(datum/source, mob/living/target, damage, list/damage_mod, list/armor_mod)
+	SIGNAL_HANDLER
+	if(!target) //Sanity
+		return
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:360]:
```suggestion
```

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 2]
```diff
@@ -371,6 +393,7 @@
 	REMOVE_TRAIT(X, TRAIT_STUNIMMUNE, RAGE_TRAIT)
 	REMOVE_TRAIT(X, TRAIT_SLOWDOWNIMMUNE, RAGE_TRAIT)
 	REMOVE_TRAIT(X, TRAIT_STAGGERIMMUNE, RAGE_TRAIT)
+	UnregisterSignal(X, COMSIG_XENOMORPH_ATTACK_LIVING) //unregister the signals; party's over
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:396]:
```suggestion
	UnregisterSignal(X, COMSIG_XENOMORPH_ATTACK_LIVING)
```

File: code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm [hunk 3]
```diff
@@ -298,6 +298,7 @@
 			charge.clear_cooldown() //Reset charge cooldown
 		if(ravage)
 			ravage.clear_cooldown() //Reset ravage cooldown
+		RegisterSignal(X, COMSIG_XENOMORPH_ATTACK_LIVING, .proc/drain_slash) //We can now regain HP by dealing damage
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm:301]:
```suggestion
		RegisterSignal(X, COMSIG_XENOMORPH_ATTACK_LIVING, .proc/drain_slash)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Comment by Carnage068** [comment]
> i don't see why this is much needed or how its balanced, rage already does alot of stuff like sunder and something, what are the needs for it ?

Piss off. Thanks.

**Comment by SplinterGP** [comment]
> 
> 
> > i don't see why this is much needed or how its balanced, rage already does alot of stuff like sunder and something, what are the needs for it ?
> 
> I think I was abundantly clear.
> 
> At the moment it is basically suicide to try to benefit from Rage at sub-zero health; this makes the risk/reward a little more viable.

i mean it can be sometimes but it also gives some big damage increase at some point, i'd say it also goes kinda backwards when the skill itself wants you to have less health in order to deal more damage and not heal it out.

**Comment by Surrealaser** [comment]
> > > i don't see why this is much needed or how its balanced, rage already does alot of stuff like sunder and something, what are the needs for it ?
> > 
> > 
> > I think I was abundantly clear.
> > At the moment it is basically suicide to try to benefit from Rage at sub-zero health; this makes the risk/reward a little more viable.
> 
> i mean it can be sometimes but it also gives some big damage increase at some point, i'd say it also goes kinda backwards when the skill itself wants you to have less health in order to deal more damage and not heal it out, so it can be counterproductive at some time.

The whole thing with Rage is a risk/reward tradeoff. Unfortunately and especially with MJP's nerf, the risk far outstrips the reward.

Life leech means the risk is ameliorated a bit, hopefully to the point of acceptability/viability, but only by aggressing.

**Pull request merged**